sklearn/utils/estimator_checks.py

```diff
@@ -1627,7 +1655,11 @@ def check_outliers_train(name, estimator_orig, readonly_memmap=True):
 def check_estimators_fit_returns_self(name, estimator_orig,
                                       readonly_memmap=False):
     """Check if self is returned when calling fit"""
-    X, y = make_blobs(random_state=0, n_samples=9, n_features=4)
+    tags = _safe_tags(estimator_orig)
+    if tags['binary_only']:
+        X, y = make_blobs(random_state=0, n_samples=9, n_features=4, centers=2)
+    else:
+        X, y = make_blobs(random_state=0, n_samples=9, n_features=4)
```

Comment:
please define `n_centers` in the `if/else` instead and use it a single call to `make_blobs`

same for the other ones

Reply:
Done